onebusaway-android/src/main/java/org/onebusaway/android/report/ui/InfrastructureIssueActivity.java

```diff
@@ -0,0 +1,443 @@
+        myIntent.putExtra(MapParams.CENTER_LON, intent.getDoubleExtra(MapParams.CENTER_LON, 0));
+        return myIntent;
+    }
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        supportRequestWindowFeature(Window.FEATURE_INDETERMINATE_PROGRESS);
+
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.infrastructure_issue);
+
+        setUpProgressBar();
+
+        setupMapFragment();
+
+        setupLocationHelper();
+
+        setupViews();
+
+        initLocation();
+    }
+
+    /**
+     * Setting up the BaseMapFragment
+     * BaseMapFragment was used to implement a map.
+     */
+    private void setupMapFragment() {
+        FragmentManager fm = getSupportFragmentManager();
+        if (mMapFragment == null) {
+            mMapFragment = new BaseMapFragment();
+            // Register listener for map focus callbacks
+            mMapFragment.setOnFocusChangeListener(this);
+            mMapFragment.setZoom(15);
+            fm.beginTransaction().add(R.id.ri_frame_map_view, mMapFragment).commit();
+        }
+        fm.beginTransaction().show(mMapFragment).commit();
+    }
+
+    /**
+     * Setting up the location helper
+     * IssueLocationHelper helps tracking the issue location and issue stop
+     */
+    private void setupLocationHelper() {
+        Location mapCenterLocation = LocationUtil.makeLocation(
+                getIntent().getDoubleExtra(MapParams.CENTER_LAT, 0),
+                getIntent().getDoubleExtra(MapParams.CENTER_LON, 0));
+
+        issueLocationHelper = new IssueLocationHelper(mapCenterLocation);
+    }
+
+    /**
+     * Initialize UI components
+     */
+    private void setupViews() {
+        addressEditText = (EditText) findViewById(R.id.ri_address_editText);
+
+        addressSearchButton = (ImageButton) findViewById(R.id.ri_search_address_button);
+        addressSearchButton.setOnClickListener(this);
+
+        addressRefreshButton = (ImageButton) findViewById(R.id.ri_refresh_address_button);
+        addressRefreshButton.setOnClickListener(this);
+
+        mainScrollView = (CustomScrollView) findViewById(R.id.ri_scrollView);
+        mainScrollView.addInterceptScrollView(findViewById(R.id.ri_frame_map_view));
+    }
+
+    /**
+     * Show the current location on the map
+     */
+    private void initLocation() {
+        syncAddress(issueLocationHelper.getIssueLocation());
+    }
+
+    @Override
+    public void onClick(View v) {
+        if (v == addressSearchButton) {
+            searchAddress();
+        } else if (v == addressRefreshButton) {
+            getAddressByLocation(issueLocationHelper.getIssueLocation());
+        }
+    }
+
+    /**
+     * Places a marker into the map from given Location
+     *
+     * @param location position for marker
+     */
+    private void updateMarkerPosition(Location location) {
+
+        Marker marker = mMapFragment.getMap().addMarker(new MarkerOptions().
```

Comment:
@barbeau I rebase the PR https://github.com/OneBusAway/onebusaway-android/issues/254 on top of this branch. I was going to implement this particular marker issue in this PR, but I thought it might be confusing. Hence, to avoid confusion for future use, I opened an issue (https://github.com/OneBusAway/onebusaway-android/issues/260) regarding this issue. I will commit a solution shortly.